[Venkat Vishwanth] iP

Vishwanth2210 opened the pull request:
## Arthur 
> “Productivity is being able to do things that you were never able to do before.” – Franz Kafka ([Source](https://dansilvestre.com/productivity-quotes/))

Arthur is your one-stop centre to help manage your tasks. Its, 
* ***Easy*** to learn 
* text-based
* ~~Fast~~ ***Super Fast*** to use

All you need to do is,
1. Download it from [here](https://github.com/Vishwanth2210/ip/releases/tag/A-Jar)
2. Double-click the jar file
3. Add your tasks
4. Enjoy your personal assistant to manage the task for you 😃 

Best of all, it's **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (Coming soon!)


Are you a Java programmer? Then good news, you can practice java too! Here's the `main` method: 
```java 
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Added Greet, Echo and Exit feature

Commit:
Added Notebook class to handle new add and list feature.

Commit:
Added Task class, Mark and Unmark feature

Commit:
Added Todo, Event and Deadline feature using Todo, Event and Deadline class

Commit:
Added semi-automated testing

Commit:
Updated Error Handling and Semi-Automated tests.

Commit:
Added delete function

Commit:
Add Storage class

This allows the tasks to be saved and read from the hard disk. It also creates the necessary files and directories.

Commit:
Add DateTime class

This allows date and time of the deadline or event task to be understood and modelled as date/time objects respectively.

Commit:
Merge branch 'branch-Level-8'

Commit:
Add more OOP features.

Closely related code are extracted and combined together in respective classes.

Commit:
Add classes into packages

Commit:
Add JUnit

Commit:
Add metadata

Commit:
Add JavaDoc and follow coding standards

Commit:
Revert "Add JavaDoc and follow coding standards"

This reverts commit 325ce5c95fa424acce34a5e333f50226fb90453a.

Commit:
Add Javadoc to methods and classes.

Commit:
Follow coding Standards

Commit:
Add find feature

Allows users to find for the tasks using keywords.

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

khoahre123 commented on src/main/java/arthur/Parser.java:
```diff
@@ -0,0 +1,77 @@
+     * @param ui The arthur.Ui class to use for user interaction
+     */
+    public void execute(TaskList taskList, Storage storage, Ui ui) {
+        String[] temp = this.command.split(" ", 2);    // Gets the first word
```
Instead of `temp`, maybe you can name the variable as `firstWord`, which is more intuitive

Vishwanth2210 replied:
Thanks for the suggestion. That is a bit better.

khoahre123 commented on src/main/java/arthur/Parser.java:
```diff
@@ -0,0 +1,77 @@
+     */
+    public void execute(TaskList taskList, Storage storage, Ui ui) {
+        String[] temp = this.command.split(" ", 2);    // Gets the first word
+        String inst = temp[0];
```
Should this variable name be clearer?

khoahre123 commented on src/main/java/arthur/TaskList.java:
```diff
@@ -0,0 +1,205 @@
+     */
+    public String listOut() {
+        StringBuilder temp = new StringBuilder("Here are the tasks in your list: \n");
+        int tempCounter = 1;
```
Maybe you can change the name to `taskCounter` so it would be more intuitive?

khoahre123 commented on src/main/java/arthur/TaskList.java:
```diff
@@ -0,0 +1,205 @@
+    public String event(String e) {
+        Event temp;
+        try {
+            String[] tempArr = e.split("/at ");
```
Perhaps use some other name besides `temp`?

khoahre123 commented on src/main/java/arthur/TaskList.java:
```diff
@@ -0,0 +1,205 @@
+     * Gives the taskList size.
+     * @return list size in int
+     */
+    public int tasksSize() {
```
You can add another verb in front of the method name so that it easier to follow, eg: `getTaskListSize`

khoahre123 commented on src/main/java/arthur/exceptions/ArthurException.java:
```diff
@@ -0,0 +1,47 @@
+        }
+
+        // Check if instructions come with description
+        if (!temp[0].equals("list") && !temp[0].equals("bye") && !temp[0].equals("find")) {
```
I noticed the same issue in several other places too, by overusing `temp`

khoahre123 commented on src/main/java/arthur/task/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+ */
+public class Deadline extends Task {
+    private final String by;
+    private final DateTime timings;
```
```suggestion
    private final DateTime timing;

```

khoahre123 commented on src/main/java/arthur/timings/DateTime.java:
```diff
@@ -0,0 +1,44 @@
+public class DateTime {
+    private LocalDate date;
+    private LocalTime time;
+    private final String str;
```
Can you explain what is this `str` used for?

Vishwanth2210 replied:
This just holds the string version of the time. At the moment it is redundant but wanted to use this in one of the features in the future.

khoahre123 reviewed (commented):
Overall, I found the code looks good except a few places overuse `temp` variable name. Also, consider splitting into smaller private method for more concise

Commit:
Merge branch 'add-gradle-support'

Commit:
Add checkstyle feature.

This helps to catch coding style violations.

Commit:
Add GUI

Commit:
Add checkstyle feature

Commit:
Merge branch 'branch-A-CheckStyle'

Commit:
Fix coding style violations

Commit:
Include JavaFX in jar

In the event the JavaFX is not installed in the user's computer,
this will ensure that the necessary dependencies are packaged
together in the jar file. This allows the user to start using the
program without the need for additional downloads.

Change the main class to the launcher class in jar

Due to the implementation of JavaFX, this will ensure that the main
class is taken as the JavaFX launcher class. This will allow the
JavaFX GUI to be utilized.

Commit:
Use assertions

At the moment, there exists some assumptions when running the
code.

These assumptions might fail if the user manages to enter the
wrong command input.

Add assert checks to ensure that assumptions are held in place.

Commit:
Improve code quality

As the code is increasing, this is necessary to ensure better
readability.

Vishwanth2210 committed:
Merge pull request #1 from Vishwanth2210/branch-A-Assertions

Use assertions

Commit:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/arthur/Storage.java

Vishwanth2210 committed:
Merge pull request #2 from Vishwanth2210/branch-A-CodeQuality

Branch a code quality

Commit:
Set up Continuous Integration (CI)

Vishwanth2210 committed:
Add reminder

Alerts the user when the task is due on that day

Vishwanth2210 committed:
Fix bugs in GUI

Bug where error messages were being displayed in the Command Line Interface has been diverted to the GUI to allow user to see the problem.

Update GUI icon, name, user icon and Arthur reply icon

Vishwanth2210 committed:
Fix bug related to date and time format

This helps to fix bug that causes program to be non-responsive when importing date and time in different format from the storage data file.

Added new GUI package.

Vishwanth2210 committed:
Add Ui screenshot

Vishwanth2210 committed:
Set theme jekyll-theme-cayman

Vishwanth2210 committed:
Update User Guide

Created a user guide.

Vishwanth2210 committed:
Merge branch 'master' of https://github.com/Vishwanth2210/ip

Vishwanth2210 committed:
Update User Guide

Vishwanth2210 committed:
Set theme jekyll-theme-cayman

Vishwanth2210 committed:
Update 'bye' command

This allows the bye command to display the final message and exit the application.

Fixed spacing between return statements.

Vishwanth2210 committed:
Merge branch 'master' of https://github.com/Vishwanth2210/ip

Vishwanth2210 committed:
Improve coding standard

This helps to improve coding standard by factoring long methods into even smaller methods.

This also converts 'magic' numbers and strings to constants.

Vishwanth2210 committed:
Fix saving the first task

Currently, saving the first task when the data file is empty causes an "Please enter a valid number!" message to be throw.

This fix will use the correct index to save the task to storage when the data file is empty, without throwing the above message.

Vishwanth2210 committed:
Update reply when list is empty

Currently, when the list is empty, Arthur still lists out the entire list.

This update now return a custom message instead of the empty list.

Fixes some misunderstanding in the comments of ArthurException and Parser java files.

Vishwanth2210 committed:
Update find command

Currently, find command is case-sensitive.

It has been updated to be case-insensitive.

The user guide has been updated to reflect the new change.

Vishwanth2210 committed:
Update user message and prefix

Currently, Arthur user visible messages that were commonly used is spread out amongst the classes. Similarly, prefix used for string manipulation is also spread out amongst classes.

With this update, all the common messages and prefix have been compiled into the Messages class and Prefix class.

Vishwanth2210 committed:
Update reminder command

Currently, when there are no reminders for that day, reminder command lists out an empty list.

This update now displays a custom message instead of an empty list.

Vishwanth2210 committed:
Update find command

Currently, find command will display an empty list when there are no matching tasks found.

With this update, a custom message will be displayed when there are no matching tasks.

Vishwanth2210 committed:
Update gradle.yml

Removes the text based UI tests as Arthur shifted to using only the GUI.

Vishwanth2210 committed:
Update Messages.java

Updates a text mistake in the mark reply template.

Vishwanth2210 committed:
Update TaskList.java

Currently, marking is done at the wrong place resulting to errors when importing tasks from storage.

This update fixes the index to be used to mark to allow correct marking.

Vishwanth2210 committed:
Update user guide

This updates the user guide to include the Ui images of Arthur.
